src/data_provider/src/sysInfoSolaris.cpp

```diff
@@ -201,9 +212,43 @@ nlohmann::json SysInfo::getPorts() const
 {
     return nlohmann::json();
 }
-void SysInfo::getProcessesInfo(std::function<void(nlohmann::json&)> /*callback*/) const
+
+void SysInfo::getProcessesInfo(std::function<void(nlohmann::json&)> callback) const
 {
-    // TODO
+    const auto procFiles{Utils::enumerateDir(WM_SYS_PROC_DIR)};
+
+    for (const auto& procFile : procFiles)
+    {
+        if (procFile.at(0) == '.')
+        {
+            continue;
+        }
+
+        std::ifstream ifPsinfoFile{WM_SYS_PROC_DIR + std::string(procFile) + "/psinfo", std::ios::binary};
+        std::ifstream ifStatusFile{WM_SYS_PROC_DIR + std::string(procFile) + "/status", std::ios::binary};
+        std::ifstream ifCredFile{WM_SYS_PROC_DIR + std::string(procFile) + "/cred", std::ios::binary};
+
+        // a relevant info is not available, get out!
+        if (!ifPsinfoFile.is_open())
+        {
+            throw std::runtime_error{"Error psinfo file not open!."};
```

Comment:
this exception is highly unlikely to occur since this file is part of the /proc FS, also how do you propose to handle this exception in this constructor?

Reply:
Ok